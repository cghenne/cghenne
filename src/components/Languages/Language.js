// @flow
import React from 'react';
import './Languages.scss';

type Props = {
  label: string,
  level: string
};

const Language = ({ label, level }: Props) => {
  return (
    <li>
      <span className="bold">{label}</span>: {level}
    </li>
  );
};

Language.displayName = 'Language';

export default Language;
