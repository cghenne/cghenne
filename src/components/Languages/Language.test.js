import React from 'react';
import { shallow } from 'enzyme';
import Language from './Language';

const fakeLanguage = {
  label: 'Chinese',
  level: 'Inexistent'
};

describe('Language', () => {
  const wrapper = shallow(<Language {...fakeLanguage} />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the language in a li tag', () => {
    expect(wrapper.find('li').exists()).toEqual(true);
  });

  it('should render the language label in bold', () => {
    expect(wrapper.find('.bold').text()).toEqual(fakeLanguage.label);
  });

  it('should render the language level', () => {
    expect(wrapper.find('li').text()).toContain(fakeLanguage.level);
  });
});
