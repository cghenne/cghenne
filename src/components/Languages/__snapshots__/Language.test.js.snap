// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language should render without crashing 1`] = `
<li>
  <span
    className="bold"
  >
    Chinese
  </span>
  : 
  Inexistent
</li>
`;
